slatedb-java/src/main/java/com/slatedb/config/SlateDBOptions.java

```diff
@@ -0,0 +1,258 @@
+ * 
+ * This class contains tuning parameters that affect database behavior,
+ * performance characteristics, and resource usage.
+ */
+public final class SlateDBOptions {
+    private final Long l0SstSizeBytes;
+    private final Duration flushInterval;
+    private final String cacheFolder;
+    private final SstBlockSize sstBlockSize;
+    private final CompactorOptions compactorOptions;
+    
+    private SlateDBOptions(Builder builder) {
+        this.l0SstSizeBytes = builder.l0SstSizeBytes;
+        this.flushInterval = builder.flushInterval;
+        this.cacheFolder = builder.cacheFolder;
+        this.sstBlockSize = builder.sstBlockSize;
+        this.compactorOptions = builder.compactorOptions;
+    }
+    
+    /**
+     * Gets the L0 SSTable size threshold in bytes.
+     * 
+     * @return the L0 SST size, or null if using default
+     */
+    public Long getL0SstSizeBytes() {
+        return l0SstSizeBytes;
+    }
+    
+    /**
+     * Gets the WAL flush interval.
+     * 
+     * @return the flush interval, or null if using default
+     */
+    public Duration getFlushInterval() {
+        return flushInterval;
+    }
+    
+    /**
+     * Gets the local cache folder path.
+     * 
+     * @return the cache folder path, or null if no caching
+     */
+    public String getCacheFolder() {
+        return cacheFolder;
+    }
+    
+    /**
+     * Gets the SSTable block size.
+     * 
+     * @return the block size, or null if using default
+     */
+    public SstBlockSize getSstBlockSize() {
+        return sstBlockSize;
+    }
+    
+    /**
+     * Gets the compactor options.
+     * 
+     * @return the compactor options, or null if using defaults
+     */
+    public CompactorOptions getCompactorOptions() {
+        return compactorOptions;
+    }
+    
+    /**
+     * Creates a new builder for SlateDBOptions.
+     * 
+     * @return a new builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    /**
+     * Creates a new builder initialized with values from this configuration.
+     * 
+     * @return a new builder instance
+     */
+    public Builder toBuilder() {
+        return new Builder()
+                .l0SstSizeBytes(l0SstSizeBytes)
+                .flushInterval(flushInterval)
+                .cacheFolder(cacheFolder)
+                .sstBlockSize(sstBlockSize)
+                .compactorOptions(compactorOptions);
+    }
+    
+    /**
+     * Creates SlateDBOptions with default settings.
+     * 
+     * @return SlateDBOptions with all defaults
+     */
+    public static SlateDBOptions defaultOptions() {
+        return builder().build();
+    }
+    
+    /**
+     * Creates SlateDBOptions optimized for high throughput.
+     * 
+     * @return SlateDBOptions for high throughput
+     */
+    public static SlateDBOptions highThroughput() {
+        return builder()
+                .l0SstSizeBytes(128L * 1024 * 1024) // 128MB
```

Comment:
Ehhh, I'm inclined to remove both of these (highThroughput/lowLatency). The defaults we provide are pretty good. For users with custom needs, I'm not sure these extra defaults provide any value.